evaluator/src/main/resources/osadl/osadl.rules.kts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /osadl/osadl.rules.kts
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.module.kotlin.readValue
+
+enum class Compatibility {
+    INHERENT,
+    YES,
+    NO,
+    UNKNOWN;
+
+    companion object {
+        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
+        @JvmStatic
+        fun fromString(value: String): Compatibility =
+            when (value) {
+                "", "Same" -> INHERENT
+                "Yes" -> YES
+                "No" -> NO
+                "?", "Unknown", "Check dependency", "Dep." -> UNKNOWN
+                else -> throw IllegalArgumentException("Unhandled compatibility '$value' found.")
+            }
+    }
+}
+
+val matrix: Map<String, Map<String, Compatibility>> = javaClass.getResourceAsStream("/osadl/osadl-matrix.json").use {
+    jsonMapper.readValue(it)
+}
+
+fun PackageRule.LicenseRule.isOsadlMatrixNonCompliant(inbound: String) =
+    object : RuleMatcher {
+        override val description = "isOsadlMatrixCompliant($license)"
+
+        override fun matches(): Boolean {
+            val outbound = license.toString()
+            return matrix.getValue(inbound).getValue(outbound) == Compatibility.NO
+        }
+    }
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    // Define a rule that is executed for each dependency.
+    dependencyRule("OSADL compliance matrix") {
```

Comment:
We have always used all caps sperate by underscoring in the names of rules, see e.g. [1](https://github.com/oss-review-toolkit/ort/blob/master/examples/evaluator-rules/src/main/resources/example.rules.kts#L98) or [2](https://github.com/oss-review-toolkit/ort/blob/master/evaluator/src/main/resources/rules/no_gpl.rules.kts#L36) so for consistency and to make easier for user understand what this rule does I propose to rename "OSADL compliance matrix" t"OSADL_LICENSE_COMPLIANCE"

Reply:
I was considering to drop the word "matrix", but then to me it sounded too much like "compliance *with* OSADL" (as if OSADL itself was a license or so). So maybe `OSADL_MATRIX_COMPLIANCE`?

Reply:
This conversation is continued [here](https://github.com/oss-review-toolkit/ort/pull/4977#discussion_r1000201431).